Shell functions

isanae opened the pull request:
- Renamed shell functions to remove redundant 'File' and 'Link'
- Added `Delete()` and `Rename()`
- Made them all return a `Result` instead of a `bool`

Al12rs commented:
To be honest I think that the redundant File and Link might have made the code more readable, but I don't care enough to tell you to revert it.

isanae commented:
To be honest, I added `shell::DeleteFile()`, then realized it's a `#define` to `DeleteFileW()` from the Windows headers, so I just called it `Delete()`, then got annoyed at the discrepancy, so I removed `File` from the rest, then got annoyed again, then removed `Link` also.

Al12rs commented:
Well it's fine
